### ISSUE TITLE
Migrate Thanos receiver to use Ketama for hashring

### ISSUE THREAD

**Issue by aarnq** [elastisys/compliantkubernetes-apps#2098]
### Description

Currently used [Hashmod is discouraged while Ketama is recommended](https://thanos.io/tip/components/receive.md/#series-distribution-algorithms), we should follow that as it should improve the experience if we need to make changes to the receive setup.

### Definition of done

- [ ] Thanos receive uses Ketama for the hashring by default
- [ ] Migration to move from Hashmod to Ketama

**Comment by aarnq** [comment]
A migration is stated in the DoD, ideally we should move all to Ketama if possible.

**Comment by aarnq** [comment]
> > A migration is stated in the DoD, ideally we should move all to Ketama if possible.
> 
> If you mean we need to migrate all environments to ketama and since ketama is the default in the upcoming apps release, wouldn't upgrade to apps 46 complete the migration stated in the DoD ?

I wrote we _should_, and it all depends on how the migration looks like which I don't really know as you asked Shafiq about testing the migration, and I assume it is pending on that if we require any special handling for it.

What is part of the DoD is that there should be migration for it, but if it isn't possible to do it in a good way then we might postpone and plan when that is done so it can be as painless as possible.

**Comment by shafi-elastisys** [comment]
No, I haven't tried Hashmod, as it's also not recommended for use. I’ve also read the Thanos documentation, and they recommend using the Ketama algorithm as well.

**Comment by aarnq** [comment]
> No, I haven't tried Hashmod, as it's also not recommended for use. I’ve also read the Thanos documentation, and they recommend using the Ketama algorithm as well.

The question is if you've tried to migrate from Hashmod to Ketama on an existing deployment.
